### ISSUE TITLE
Mandrel for JDK 21: native-image --enable-preview fails

### ISSUE THREAD

**Issue by leahneukirchen** [graalvm/mandrel#658]
## Description

The JDK 21 included in Graal 23 has support for `--enable-preview`, but `native-image` lacks the "preview modules" to use them.

## How To Reproduce

```
% pwd
/tmp/mandrel-java21-23.1.1.0-Final

% cat Foo.java
void main() {
  System.out.println("yay");
}

% ./bin/javac --enable-preview --release 21 Foo.java
Note: Foo.java uses preview features of Java SE 21.
Note: Recompile with -Xlint:preview for details.

% ./bin/java --enable-preview Foo           
yay

% ./bin/native-image --enable-preview Foo           
Error: The directory containing the preview modules was not found.
```

### Configuration:
 - OS: Void Linux
 - Architecture: x86_64
 - Mandrel version: 23.1.1.0; official Mandrel binaries
 - JDK version: openjdk 21.0.1 2023-10-17 LTS; official Mandrel binaries

## Expected behavior

I expect the program to be compiled into a native image and then behave as if the JVM ran it.

## Additional context

A `svm-preview` directory seems to be lacking, but I'm not sure where it could come from.

With `graalvm-community-openjdk-21.0.1+12.1` this works.

**Comment by jerboaa** [comment]
Thanks for the report. We'll look into it.

**Comment by jerboaa** [comment]
Confirmed. The mandrel build seems to be missing the `svm-foreign.jar`:

GraalVM Community for JDK 21
```
$ ls graalvm-community-openjdk-21+35.1/lib/svm-preview/builder/
svm-foreign.jar
```

Mandrel for JDK 21:

```
$ ls mandrel-java21-23.1.0.0-Final/lib/svm-preview/builder
ls: cannot access 'mandrel-java21-23.1.0.0-Final/lib/svm-preview/builder': No such file or directory
```

Removing the `svm-preview` folder from the GraalVM community distro results in the same error as reported.

We'll fix it for the January release (`23.1.2.0-Final`).

**Comment by jerboaa** [comment]
Fix has been merged. I'll close this now. Please re-open if the release later this month won't have this fixed.
